harness/determined/common/experimental/trial.py

```diff
@@ -289,28 +290,46 @@ def __repr__(self) -> str:
         return "Trial(id={})".format(self.id)
 
     def stream_metrics(self, group: str) -> Iterable[metrics.TrialMetrics]:
+        """Get a stream of metrics for this trial.
+
+        Arguments:
+            group: The metric group to stream.  Valid values are "validation" and "training".
+
+        Returns:
+            An iterable of :class:`~determined.experimental.TrialMetrics` objects.
         """
-        Streams metrics for this trial sorted by
-        trial_id, trial_run_id and steps_completed.
-        """
+        if group not in ("training", "validation"):
+            raise ValueError(
+                f"Invalid metric group: {group}. Valid values are 'training' and 'validation'"
+            )
```

Comment:
We shouldn't be enforcing this. It would break inference metrics

Reply:
For reference: https://docs.determined.ai/latest/model-dev-guide/api-guides/batch-processing/batch-process-api-ug.html#step-5-send-and-query-custom-inference-metrics-optional

Here we are advising users to utilize the new generic metrics for inference.

What I would say is that the reason we kept the `stream_*_metrics` stuff was because users were already using it and it's a level of convenience. But we don't want to block them from searching by the generic metrics

Reply:
Awesome. Thanks for the feedback. I took my best guess.

Will fix.